packages/react-components/src/components/TypingIndicator.tsx

```diff
@@ -56,57 +52,85 @@ const getDefaultComponents = (
     const additionalCharCount =
       2 * (countOfUsersMentioned - 1) + (typingUser.displayName ? typingUser.displayName.length : 0);
     if (totalCharacterCount + additionalCharCount <= MAXIMUM_LENGTH_OF_TYPING_USERS || countOfUsersMentioned === 1) {
-      typingUsersMentioned.push(typingUser);
+      if (typingUser.displayName) {
+        typingUsersMentioned.push(typingUser.displayName);
+      }
       totalCharacterCount += additionalCharCount;
       countOfUsersMentioned += 1;
     } else {
       break;
     }
   }
 
-  typingUsersMentioned.forEach((typingUser, index) => {
-    displayComponents.push(
-      <span
-        className={mergeStyles(typingIndicatorListStyle, styles?.typingUserDisplayName)}
-        key={`typing indicator display string ${index}`}
-      >
-        {index < typingUsers.length - 1 ? typingUser.displayName + ', ' : typingUser.displayName}
-      </span>
-    );
-  });
-
   const countOfUsersNotMentioned = typingUsers.length - typingUsersMentioned.length;
-  if (countOfUsersNotMentioned > 0) {
-    displayComponents.push(
-      <span className={mergeStyles(typingIndicatorVerbStyle, styles?.typingString)} key={'UsersCountKey'}>
-        {` and ${countOfUsersNotMentioned} other${countOfUsersNotMentioned === 1 ? '' : 's'}`}
-      </span>
-    );
+
+  let typingString = '';
+  if (typingUsers.length === 1) {
+    typingString = strings.typing_indicator_singular;
+  } else if (typingUsers.length > 1 && countOfUsersNotMentioned === 0) {
+    typingString = strings.typing_indicator_plural;
+  } else if (typingUsers.length > 1 && countOfUsersNotMentioned === 1) {
+    typingString = strings.typing_indicator_shortened_plural;
+  } else if (typingUsers.length > 1 && countOfUsersNotMentioned > 1) {
+    typingString = strings.typing_indicator_shortened_plural_2;
   }
 
-  return displayComponents;
+  return (
+    <Stack horizontal className={mergeStyles(typingIndicatorStringStyle, styles?.typingString)} key="typingStringKey">
+      {formatElements(typingString, {
+        users: onRenderUsers ? (
+          onRenderUsers(typingUsers)
+        ) : (
+          <Stack className={mergeStyles(typingIndicatorStringStyle, styles?.typingUserDisplayName)}>
+            {typingUsersMentioned.join(', ')}
+          </Stack>
+        ),
+        numOthers: <>`${countOfUsersNotMentioned}`</>
+      })}
+    </Stack>
+  );
 };
 
-const defaultTypingString = (typingUsers: CommunicationParticipant[]): string => {
-  return typingUsers.length > 0 ? (typingUsers.length > 1 ? ' are typing...' : ' is typing...') : '';
+const formatElements = (str: string, vars: IObjectMap<JSX.Element>): JSX.Element[] => {
```

Comment:
What is this function doing? This needs renamed as `formatElements` is far too generic a name.